src/components/d3/CirclesGroup.js

```diff
@@ -0,0 +1,69 @@
+const React = require('react');
+
+const StyleConstants = require('../../constants/Style');
+
+const CirclesGroup = React.createClass({
+  propTypes: {
+    adjustedHeight: React.PropTypes.number.isRequired,
+    circleRadius: React.PropTypes.number,
+    data: React.PropTypes.array.isRequired,
+    shouldAnimate: React.PropTypes.bool,
+    translation: React.PropTypes.string,
+    xScaleValueFunction: React.PropTypes.func.isRequired,
+    yScaleValueFunction: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps () {
+    return {
+      circleRadius: 3,
+      shouldAnimate: true,
+      translation: 'translate(0,0)'
+    };
+  },
+
+  componentDidMount () {
+    this._animateCircles();
+  },
+
+  componentDidUpdate () {
+    this._animateCircles();
+  },
+
+  _animateCircles () {
+    if (this.props.shouldAnimate) {
```

Comment:
Same here. Did you want to make it consistent with the `render` function? I'm fine either way where you have the `this.props` assigned to variables at the top.